internal/resource/config/authenticationapi/application/authentication_api_application_resource.go

```diff
@@ -0,0 +1,269 @@
+// GetSchema defines the schema for the resource.
+func (r *authenticationApiApplicationResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	schema := schema.Schema{
+		Description: "Manages an Authentication Api Application",
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Description: "The Authentication API Application Name. Name must be unique.",
+				Required:    true,
+			},
+			"url": schema.StringAttribute{
+				Description: "The Authentication API Application redirect URL.",
+				Required:    true,
+				Validators: []validator.String{
+					configvalidators.ValidUrl(),
+				},
+			},
+			"description": schema.StringAttribute{
+				Description: "The Authentication API Application description.",
+				Optional:    true,
+			},
+			"additional_allowed_origins": schema.ListAttribute{
+				Description: "The domain in the redirect URL is always whitelisted. This field contains a list of additional allowed origin URL's for cross-origin resource sharing.",
+				Computed:    true,
+				Optional:    true,
+				Default:     listdefault.StaticValue(emptyStringList),
+				ElementType: types.StringType,
+				Validators: []validator.List{
+					listvalidator.SizeAtLeast(1),
+					listvalidator.ValueStringsAre(
+						configvalidators.ValidUrl(),
+					),
+				},
+			},
+			"client_for_redirectless_mode_ref": schema.SingleNestedAttribute{
+				Optional:    true,
+				Computed:    true,
+				Required:    false,
+				Default:     objectdefault.StaticValue(types.ObjectNull(resourcelink.AttrType())),
+				Description: "The client this application must use if it invokes the authentication API in redirectless mode. No client may be specified if restrictAccessToRedirectlessMode is false under /authenticationApi/settings.",
+				Attributes:  resourcelink.ToSchema(),
+			},
+		},
+	}
+
+	id.ToSchema(&schema)
+	id.ToSchemaCustomId(&schema,
+		"application_id",
+		true,
+		"The persistent, unique ID for the Authentication API application. It can be any combination of [a-zA-Z0-9._-]. This property is system-assigned if not specified.",
+	)
+	resp.Schema = schema
+}
+
+// Metadata returns the resource type name.
+func (r *authenticationApiApplicationResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_authentication_api_application"
+}
+
+func (r *authenticationApiApplicationResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func addOptionalAuthenticationApiApplicationFields(ctx context.Context, addRequest *client.AuthnApiApplication, plan authenticationApiApplicationModel) error {
+
+	if internaltypes.IsDefined(plan.Description) {
```

Comment:
Don't need isDefined checks for simple attributes like these (strings, ints, etc). You might not need it either for the set/list below either, though I'm not sure on that.